WooCommerce/Classes/POS/ViewModels/ItemSelectorViewModel.swift

```diff
@@ -23,26 +39,26 @@ final class ItemSelectorViewModel: ObservableObject {
 
     @MainActor
     func populatePointOfSaleItems() async {
-        isSyncingItems = true
         do {
             items = try await itemProvider.providePointOfSaleItems()
+            state = .loaded
         } catch {
             DDLogError("Error on load while fetching product data: \(error)")
+            state = .error
         }
-        isSyncingItems = false
     }
 
     @MainActor
     func reload() async {
-        isSyncingItems = true
```

Comment:
Shouldnt we have `state = .loading` here?
Or maybe set it in case `items` is empty?

Reply:
Yep! Same as https://github.com/woocommerce/woocommerce-ios/pull/13289#discussion_r1671884821